hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/test/java/org/apache/hadoop/mapreduce/TestJobSubmissionFiles.java

```diff
@@ -139,4 +141,19 @@ public void testGetStagingDirWhenShortFileOwnerNameAndShortUserName()
     assertEquals(stagingPath,
         JobSubmissionFiles.getStagingDir(cluster, conf, user));
   }
+
+  @Test
+  public void testDirPermission() throws Exception {
```

Comment:
Does this test actually cover the bug? Should the test be configuring umask to 700 to simulate the bug more accurately?

Ideally, this should be a test that 1) fails before applying the `JobSubmissionFiles` patch, and 2) starts passing after applying the `JobSubmissionFiles` patch.